tendermint-rs/src/amino_types/proposal.rs

```diff
@@ -155,6 +155,13 @@ impl SignableMsg for SignProposalRequest {
             None => None,
         }
     }
+
+    fn height(&self) -> Option<i64> {
+        if let Some(proposal) = &self.proposal {
+            return Some(proposal.height);
+        }
+        None
```

Comment:
Idiomatic way to write this is probably: `self.proposal.as_ref().map(|proposal| proposal.height)`